### PR TITLE
feat(ST-63): add remove student from classroom api

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added DELETE `classrooms/enroll` API route 
- Added a function called removeStudentFromClassroom
- Checked the classroom does belongs to the current user or not
- Added a custom deleteOne method inside ClassroomsStudentsRepository
- Returned meaningful messages

## Associated Ticket Link(s)

- https://trello.com/c/KMpZ8W0G

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/classrooms/classrooms.service.ts [hunk 1]
```diff
@@ -68,4 +68,22 @@ export class ClassroomsService {
...
   }
+
+  async removeStudentFromClassroom(userId: number, deleteEnrollDto: EnrollStudentDto) {
+    const { studentId, classroomId } = deleteEnrollDto;
+
+    const user = await this.usersRepository.findOneOrFail({ id: userId });
+    const classroom = await this.classroomsRepository.findOneOrFail({ id: classroomId });
+
+    if (user.teacher.id !== classroom.teacher.id) {
+      throw new UnauthorizedException("You can't remove students from other's classroom!");
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:79]: [NIT] - I think the UnauthorizeException gives too much information. I think it's better to just let the teacher know that they are the not the teacher of the classroom.

File: src/classrooms-students/classrooms-students.repository.ts [hunk 2]
```diff
@@ -5,4 +5,9 @@ import { EntityRepository } from "@mikro-orm/postgresql";
 import { ClassroomStudent } from "@/common/entities/classrooms-students.entity";
 
 @Injectable()
-export class ClassroomsStudentsRepository extends EntityRepository<ClassroomStudent> {}
+export class ClassroomsStudentsRepository extends EntityRepository<ClassroomStudent> {
+  async deleteOne(clssroomStudent: ClassroomStudent) {
+    await this.em.removeAndFlush(clssroomStudent);
+    return { message: "Removed Student from Classroom!" };
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms-students/classrooms-students.repository.ts:12]: Can remove this method. You can just call removeAndFlush in the service layer.

File: src/classrooms-students/classrooms-students.repository.ts [hunk 3]
```diff
@@ -5,4 +5,9 @@ import { EntityRepository } from "@mikro-orm/postgresql";
 import { ClassroomStudent } from "@/common/entities/classrooms-students.entity";
 
 @Injectable()
-export class ClassroomsStudentsRepository extends EntityRepository<ClassroomStudent> {}
+export class ClassroomsStudentsRepository extends EntityRepository<ClassroomStudent> {
+  async deleteOne(clssroomStudent: ClassroomStudent) {
+    await this.em.removeAndFlush(clssroomStudent);
+    return { message: "Removed Student from Classroom!" };
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms-students/classrooms-students.repository.ts:11]: Best to handle notifications to user in client side

File: src/classrooms/classrooms.controller.ts [hunk 4]
```diff
@@ -33,4 +35,14 @@ export class ClassroomsController {
   addStudent(@Body() enrollStudentDto: EnrollStudentDto) {
     return this.classroomsService.addStudent(enrollStudentDto);
   }
+
+  @Delete("enroll")
```
Comment (ragib-hossain-sazim) [inline src/classrooms/classrooms.controller.ts:39]: is it the endpoint name to remove students?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Pull request merged**